From 32fcef919a6db30832af8e09eb6d35d5a869b795 Mon Sep 17 00:00:00 2001
From: Volodymyr Babchuk <volodymyr_babchuk@epam.com>
Date: Thu, 11 Nov 2021 23:20:50 +0200
Subject: [PATCH] [HACK] s4: boot U-Boot in EL2 mode

Xen requires EL2 mode, so we need to start U-Boot in this mode.

Signed-off-by: Volodymyr Babchuk <volodymyr_babchuk@epam.com>
---
 plat/renesas/rcar_gen4/bl31_plat_setup.c | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/plat/renesas/rcar_gen4/bl31_plat_setup.c b/plat/renesas/rcar_gen4/bl31_plat_setup.c
index 3eb62cf2d..2e087f77d 100644
--- a/plat/renesas/rcar_gen4/bl31_plat_setup.c
+++ b/plat/renesas/rcar_gen4/bl31_plat_setup.c
@@ -62,6 +62,11 @@ struct entry_point_info *bl31_plat_get_next_image_ep_info(uint32_t type)
 	next_image_info = (type == NON_SECURE) ?
 		&from_bl2->bl33_ep_info : &from_bl2->bl32_ep_info;
 
+	/* HACK: Boot U-Boot in EL2. This should be configured from IPL side */
+	if (type == NON_SECURE)
+	    next_image_info->spsr = SPSR_64(MODE_EL2, MODE_SP_ELX,
+					    DISABLE_ALL_EXCEPTIONS);
+
 	return (next_image_info->pc != 0U) ? next_image_info : NULL;
 }
 
-- 
2.17.1

